lib/src/repo.rs

```diff
@@ -1855,6 +1854,15 @@ pub enum EditCommitError {
     BackendError(#[from] BackendError),
 }
 
+impl From<RewriteRootCommit> for EditCommitError {
```

Comment:
This could be handled by thiserror:
```rust
    #[error(transparent)]
    RewriteRootCommit(#[from] RewriteRootCommit),
```

I think https://github.com/jj-vcs/jj/pull/5623#issuecomment-2645992444 is also a good option. I didn't know the match pattern `Foo` is (roughly?) equivalent to `Foo { .. }`.

Reply:
> I didn't know the match pattern `Foo` is (roughly?) equivalent to `Foo { .. }`.

I believe what actually happens is that `Foo` is parsed as a unit struct pattern if there is such a unit struct, but if there is not, then it’s a variable name. So it changes syntactic form depending on the shape of the type in scope, which is weird. `Foo {}` should avoid that, since it can only parse as a struct pattern. (Conversely, `Foo { .. }` would fail if `Foo` was removed or became something else that can’t match that way, I think.)

Reply:
I also didn't realize `Foo {}` and `Foo` were equivalent for empty structs. It would be better if only `Foo {}` worked I think to avoid this entire issue.

I thought about using thiserror, wasn't sure why it wasn't used in the first place. I guess the downside is that the `EditCommitError::RewriteRootCommit` enum will have an empty struct in it.

Overall I think it's better to handle this with `From`, either using thiserror or by hand, since hopefully avoids "bad" `map_err` calls being added in the future. I don't have an opinion on which is better though. Whatever you prefer is fine by me.

Reply:
I don't have strong feeling. There's only one caller of `From`, so I would choose anything that's easier to implement (i.e. `#[from]` or `.map_err()`.)